pkg/cloudprovider/providers/vsphere/vclib/datacenter.go

```diff
@@ -21,7 +21,7 @@ type Datacenter struct {
 
 // GetDatacenter returns the DataCenter Object for the given datacenterPath
 // If datacenter is located in a folder, include full path to datacenter else just provide the datacenter name
-func GetDatacenter(ctx context.Context, connection VSphereConnection, datacenterPath string) (*Datacenter, error) {
+func GetDatacenter(ctx context.Context, connection *VSphereConnection, datacenterPath string) (*Datacenter, error) {
```

Comment:
This will be called by cloud provider
`vclib.GetDatacenter(ctx, &vSphereConnection, DatacenterName)`